miso-web/src/main/webapp/scripts/hot_sample.js

```diff
@@ -263,7 +263,13 @@ HotTarget.sample = (function() {
               flat.externalName = sam.externalName;
             },
             pack : function(sam, flat, errorHandler) {
-              sam.externalName = flat.externalName;
+              if (flat.potentialIdentities) {
+                var selectedIdentity = Utils.array.findFirstOrNull(function(item) {
+                  return item.label == flat.identityAlias}, flat.potentialIdentities);
+                if (!selectedIdentity) {
+                  sam.externalName = flat.externalName;
+                }
+              }
```

Comment:
It should probably just trigger the error handler and give up. There's a `;` missing after `identityAlias` on 268. This function and the other pack function are almost the same and set duplicate fields in the sample. I think it would be best to collapse them in a function like:
```
var getIdentityProperty = function(flat, property) {
return Utils.array.maybeGetProperty(Utils.array.findFirstOrNull(function(item) {
            return item.label == flat.identityAlias;
}, flat.potentialIdentities || []), property);
}
```

And then consume it as:
`sam.parentAlias = getIdentityProperty(flat, 'parentAlias');` and `sam.externalName = getIdentityProperty(flat, 'externalName');`